iorgen/parser_mtlg.py

```diff
@@ -0,0 +1,372 @@
+        return decl + 'end'
+
+    def get_fun_prototype(self, name: str, args: List[Variable]) -> str:
+        """Returns the Metalang base function prototype"""
+        name = var_name(name)
+        if not args:
+            return 'def void {}()'.format(name)
+
+        arg_list = self.get_var_call(args[0])
+
+        for arg in args[1:]:
+            arg_list += ', {}'.format(self.get_var_call(arg))
+
+        return 'def void {}({})'.format(name, arg_list)
+
+    def get_fun_decl(self, name: str, args: List[Variable]) -> str:
+        """Returns the Metalang base function declaration"""
+        fun = ''
+        for arg in args:
+            com = self.get_param_comment(arg)
+            if com:
+                fun += com + '\n'
+        com = ''
+        if self.input_data.output:
+            com = self.get_fun_comment(self.input_data.output) + '\n'
+        return fun + '{}\n{}end'.format(self.get_fun_prototype(name, args), com)
+
+    def get_fun_call(self, name: str, args: List[Variable], indent: int = 0) -> str:
+        """Return the Metalang function call"""
+        name = var_name(name)
+        if not args:
+            return '{}{}()'.format(" " * indent, name)
+
+        arg_list = (var_name(args[0].name))
+
+        for arg in args[1:]:
+            arg_list += ', {}'.format(var_name(arg.name))
+
+        return '{}{}({})'.format(" " * indent, name, arg_list)
+
+    def is_native_type(self, type_: Type) -> bool:
+        """Return true if type is int or char"""
+        return type_.main == TypeEnum.INT or type_.main == TypeEnum.CHAR
+
+    def get_var_call(self, var: Variable) -> str:
+        """Return a Metalang variable call"""
+        return '{} {}'.format(self.get_type_str(var.type), var_name(var.name))
+
+
+    def is_list_in_stdlib(self, var: Variable) -> bool:
+        """Tells if there is a function in stdlib to parse string"""
+        return var.type.encapsulated.main == TypeEnum.LIST \
+               or self.is_native_type(var.type.encapsulated)
+
+    def get_var_decl(self, var: Variable, indent: int = 0) -> str:
+        """Return declaration of a variable in Metalang"""
+
+        str_indent = " " * indent
+        # hack because parsing lists needs a loop for most cases
+        if var.type.main == TypeEnum.CHAR:
+            decl = self.get_char_decl(var, indent)
+        elif var.type.main == TypeEnum.LIST \
+             and not self.is_list_in_stdlib(var):
+            decl = '{}def {}{}'.format(str_indent, self.get_var_call(var),
+                                       self.get_list_parsing(var.type, indent))
+        elif var.type.main == TypeEnum.STRUCT:
+            decl = self.get_struct_parsing(var, indent)
+        else:
+            decl = '{}def {} = {}'.format(str_indent, self.get_var_call(var),
+                                          self.get_type_parsing(var.type))
+
+        com = ''
+        if var.comment:
+            com = ' ' + self.get_var_comment(var)
```

Comment:
I’m seriously considering removing the comments of variables in the main function. But with metalang it renders very badly.
It would be better to remove this line.